app/bundles/ReportBundle/Assets/js/report.js

```diff
@@ -383,3 +383,86 @@ Mautic.getHighestIndex = function (selector) {
 
     return parseInt(highestIndex);
 };
+
+Mautic.cloneReportRow = function (containerId) {
+    // Get the existing filter container
+    var container = mQuery('#' + containerId);
+
+    // Extract values from the existing filter
+    var glue = container.find('.filter-glue').val();
+    var column = container.find('.filter-columns').val();
+
+    // Handle the value field, considering different input types
+    var valueInput = container.find('.filter-value');
+    var value;
+    if (valueInput.is('select')) {
+        value = valueInput.val();
+    } else if (valueInput.is('input')) {
+        value = valueInput.val();
+    } else if (valueInput.is(':radio')) {
+        value = container.find('.filter-value:checked').val();
+    } else {
+        value = valueInput.val();
+    }
+
+    // Get the dynamic option
+    var dynamic = container.find('[name*="[dynamic]"]:checked').val();
+
+    // Add a new filter row using the existing add function
+    Mautic.addReportRow('report_filters');
+
+    // Get the new container by finding the last filter container
+    var newContainer = mQuery('#report_filters').find('> .panel.in-group').last();
+
+    // Extract index from new container ID
+    var newContainerId = newContainer.attr('id');
+    var idParts = newContainerId.split('_');
+    var index = idParts[2];
```

Comment:
This var doesn't seem to be used. And when you remove this, you can remove the 3 lines above as well.

Reply:
Hm, I had a lot of trouble to find a functional solution, maybe the code got a bit verbose at the end

would you like to directly edit the PR and commit the changes? It’d be perfect

Reply:
Done